### PR TITLE
SPM-1089: Fix setting of thid_party flag for repos

### PR DESCRIPTION
**Pull request by semtexzv**
## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [x] Input Validation
- [x] Output Encoding
- [x] Authentication and Password Management
- [x] Session Management
- [x] Access Control
- [x] Cryptographic Practices
- [x] Error Handling and Logging
- [x] Data Protection
- [x] Communication Security
- [x] System Configuration
- [x] Database Security
- [x] File Management
- [x] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by Josca — CHANGES_REQUESTED** [review]
1) Tests fail:
~~~
 zookeeper    | ===> Check if /var/lib/zookeeper/log is writable ...
test         | 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100  118k    0 36708  100 84892  86780   195k --:--:-- --:--:-- --:--:--  280k
test         | 1762,1764c1762
test         | <             }
test         | <         },
test         | <         "/api/patch/v1/systems/{inventory_id}/": {
test         | ---
test         | >             },
test         | docs/openapi.json different from file generated with './scripts/generate_docs.sh'!
test exited with code 1
~~~
2) Please associate PR (and commit) to some ticket.

File: vmaas_sync/repo_sync.go [hunk 1]
```diff
@@ -7,18 +7,30 @@ import (
...
-	if len(updateRepos) == 0 {
+	if len(redhatRepos) == 0 && len(thirdpartyRepos) == 0 {
 		return nil
 	}
 
-	err = database.Db.Exec("UPDATE repo SET third_party = false WHERE name in (?)", updateRepos).Error
+	err = database.Db.Exec("UPDATE repo SET third_party = false WHERE name in (?)", redhatRepos).Error
 	if err != nil {
-		return errors.WithMessage(err, "Updating repo third_party flag")
+		return errors.WithMessage(err, "Updating repo third_party flag for redhat contet")
```
Comment (Josca) [inline vmaas_sync/repo_sync.go:21]: typo

File: vmaas_sync/repo_sync.go [hunk 2]
```diff
@@ -7,18 +7,30 @@ import (
...
 	}
+
+	err = database.Db.Exec("UPDATE repo SET third_party = true WHERE name in (?)", thirdpartyRepos).Error
```
Comment (Josca) [inline vmaas_sync/repo_sync.go:24]: Can not you just set third_party=true all not Red Hat repos? You can remove one exec command then.

**Review by Josca — CHANGES_REQUESTED** [review]

File: vmaas_sync/repo_sync.go (hunk 1 above)
Comment (semtexzv) [reply vmaas_sync/repo_sync.go]: Fixed

File: vmaas_sync/repo_sync.go (hunk 2 above)
Comment (semtexzv) [reply vmaas_sync/repo_sync.go]: Fixed

**Review by Josca — APPROVED** [review]
Great, I've tested that in CI and it works. EPEL repos are updated to third_party=true.

**Pull request merged**